Python/_warnings.c

```diff
@@ -73,8 +73,12 @@ create_filter(PyObject *category, _Py_Identifier *id, const char *modname)
     }
 
     /* This assumes the line number is zero for now. */
-    return PyTuple_Pack(5, action_str, Py_None,
-                        category, modname_obj, _PyLong_GetZero());
+    res = PyTuple_Pack(5, action_str, Py_None,
+                       category, modname_obj, _PyLong_GetZero());
+    if (modname != NULL) {
+        Py_DECREF(modname_obj);
```

Comment:
This is not correct when modname_obj is None. By the way, modname cannot be NULL at this point.

I wrote PR #23365 to fix this leak.

Reply:
Please look again. ``modname`` can be NULL here.

```
    if (modname != NULL) {
        modname_obj = PyUnicode_InternFromString(modname);
        if (modname_obj == NULL) {
            return NULL;
        }
    } else {
        modname_obj = Py_None;
    }

        if (modname != NULL) {
        Py_DECREF(modname_obj);
    }
```

Reply:
Oh, I was confused between "modname" and "modname_obj".

Reply:
I would prefer to more regular code like https://github.com/python/cpython/pull/23365/files rather than checking variable A to decide what to do with variable B.

Reply:
I have merged your PR and backed my changes out. Your code is easier to understand.